.github/workflows/linting.yml

```diff
@@ -0,0 +1,38 @@
+name: Linting
+
+on:
+  workflow_call:
+
+jobs:
+  linting:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Detect File Changes
+        uses: dorny/paths-filter@v2
+        id: changed
+        with:
+          list-files: shell
+          filters: |
+            wpcontent:
+              - added|modified: 'wp-content/plugins/core/**/*.{pcss,json,js}'
```

Comment:
We'll want support for changes to config files in the project root as those get linted as well. `- added|modified: *.{json,js}`